### PR TITLE
Refactor and add customer filtering

### PR DESCRIPTION
**Pull request by islean**
## Description

There are multiple filters where the naming is unclear, mainly about customer vs customer_id and singular vs plural. This PR introduces some new filters and refactors some existing methods to make the names a bit more truthful.

### Added

- 

### Changed

-

### Fixed

- Naming of sample and pool filtering based on customers

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/store/filters/status_pool_filters.py [hunk 1]
```diff
@@ -71,13 +71,20 @@ def filter_pools_by_customer(pools: Query, customer: Customer, **kwargs) -> Quer
     return pools.filter(Pool.customer == customer)
 
 
+def filter_pools_by_customers(pools: Query, customers: list[Customer], **kwargs) -> Query:
+    """Return pools by customers."""
+    customer_ids = [customer.id for customer in customers]
+    return pools.filter(Pool.customer_id.in_(customer_ids))
```
Comment (islean) [inline cg/store/filters/status_pool_filters.py:77]: Not beautiful but apparently there is no support for an "_in" on relationships, meaning that one needs to go from having `Customer` objects to the ids.

File: cg/store/filters/status_sample_filters.py [hunk 2]
```diff
@@ -147,6 +147,12 @@ def filter_samples_by_customer(samples: Query, customer: Customer, **kwargs) ->
     return samples.filter(Sample.customer == customer)
 
 
+def filter_samples_by_customers(samples: Query, customers: list[Customer], **kwargs) -> Query:
+    """Return samples by customers."""
+    customer_ids = [customer.id for customer in customers]
+    return samples.filter(Sample.customer_id.in_(customer_ids))
```
Comment (islean) [inline cg/store/filters/status_sample_filters.py:153]: Same thing here, not beautiful but no support for `in` with relationships.

**Review by ChrOertlin — APPROVED** [review]
looks good to me

**Comment by islean** [comment]
Tested on `cg-services-stage` where setting my user to be not admin and changing the customer_user entries changed the responses in the cURLs.

**Comment by islean** [comment]
Deployed to the VMs

**Pull request merged**